File: MenaCart/Domain/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Models
{
    public enum ApprovalStatus
    {
        Pending,
        Approved,
        Rejected
    }

    public class Product
    {
        [Key]
        public int ProductId { get; set; }

        [Required]
        public int SellerId { get; set; }

        [Required]
        public int CategoryId { get; set; }

        [Required]
        [MaxLength(200)]
        public string Name { get; set; }

        public string Description { get; set; }

        [Column(TypeName = "decimal(10,2)")]
        public decimal BasePrice { get; set; }

        [MaxLength(100)]
        public string Brand { get; set; }

        [Required]
        public ApprovalStatus ApprovalStatus { get; set; } = Models.ApprovalStatus.Pending;

        [Column(TypeName = "decimal(3,2)")]
        public decimal AverageRating { get; set; } = 0;

        public int ReviewCount { get; set; } = 0;

        public bool IsActive { get; set; } = true;

        [MaxLength(500)]
        public string? RejectionReason { get; set; }

        [MaxLength(500)]
        public string? MainImageUrl { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        [ForeignKey(nameof(SellerId))]
        public SellerProfile SellerProfile { get; set; }

        [ForeignKey(nameof(CategoryId))]
        public Category Category { get; set; }

        public ICollection<ProductImage> ProductImages { get; set; }
        public ICollection<ProductVariant> ProductVariants { get; set; }
        public ICollection<Review> Reviews { get; set; }
    }
}

File: menacartfrontend/src/types/product.ts
export interface ProductVariant {
  variantId: number;
  sku: string;
  color: string | null;
  size: string | null;
  stockQuantity: number;
  price: number;
  mainImageUrl: string | null;
  variantImages: string[];
}

export interface Product {
  productId: number;
  name: string;
  description: string | null;
  basePrice: number;
  brand: string | null;
  approvalStatus: string;
  averageRating: number;
  reviewCount: number;
  isActive: boolean;
  rejectionReason: string | null;
  mainImageUrl: string | null;
  categoryId: number;
  categoryName: string;
  sellerId: number;
  storeName: string;
  createdAt: string;
  productImages: string[];
  variants: ProductVariant[];
}

export interface CreateProductRequest {
  name: string;
  description?: string;
  basePrice: number;
  brand?: string;
  categoryId: number;
  mainImageUrl?: string;
  productImages?: string[];
  variants: CreateVariantRequest[];
}

export interface CreateVariantRequest {
  sku: string;
  color?: string;
  size?: string;
  stockQuantity: number;
  price: number;
  mainImageUrl?: string;
  variantImages?: string[];
}

export interface UpdateProductRequest {
  name?: string;
  description?: string;
  basePrice?: number;
  brand?: string;
  categoryId?: number;
  mainImageUrl?: string;
  productImages?: string[];
  variants?: UpdateVariantRequest[];
}

export interface UpdateVariantRequest {
  variantId?: number;
  sku?: string;
  color?: string;
  size?: string;
  stockQuantity?: number;
  price?: number;
  mainImageUrl?: string;
  variantImages?: string[];
}

export interface ApproveProductRequest {
  status: 'Approved' | 'Rejected';
  rejectionReason?: string;
}

export interface ProductBrowseParams {
  search?: string;
  categoryId?: number;
  sellerId?: number;
  page?: number;
  pageSize?: number;
}
